### PR TITLE
VAS migration - splitted

### PR DESCRIPTION
**Pull request by vguzmanp**
Same changes as  https://github.com/LongoMatch/longomatch/pull/93 but splitted to avoid GitHub limits

### CODE REVIEW

File: LongoMatch.Core/StyleConf.cs [hunk 1]
```diff
@@ -1,230 +0,0 @@
...
-
-		public const int ButtonHeaderHeight = 18;
-		public const int ButtonHeaderWidth = 18;
-		public const int ButtonRecWidth = 40;
-		public const int ButtonLineWidth = 2;
-		public const int ButtonHeaderFontSize = 14;
-		public const int ButtonNameFontSize = 18;
-		public const int ButtonTimerFontSize = 24;
-		public const int ButtonButtonsFontSize = 10;
-		public const int ButtonMinWidth = 100;
```
Comment (vguzmanp) [inline LongoMatch.Core/StyleConf.cs]: StyleConf will be different in each application, we will need to split it.

File: longomatch-mobile [hunk 2]
```diff
@@ -0,0 +1 @@
+Subproject commit ec07512959591d8d0eb922876e3be67dfada19ef
```
Comment (vguzmanp) [inline longomatch-mobile]: We shouldn't have a submodule here for this

File: share/longomatch [hunk 3]
```diff
@@ -0,0 +1 @@
+/Users/vguzman/Development/longomatch/data/
```
Comment (vguzmanp) [inline share/longomatch]: ups :D

**Comment by ylatuya** [comment]
Rename class and file:

- ActionLinkLongoMatch
- DashboardButtonLongoMatch
- PlayerLongoMatch
- ProjectLongoMatch
- TimelineEventLongoMatch
- TimerLongoMatch

Rename file:

- EventTypeLongoMatch
- ConfigLongoMatch

For latter:

Split Config into:

- App: store aplication states, default directories, etc...
- Config: stores application configuration.
- Move some constants such as SoftwareName in a way that they can be easilly configured by the apllication, maybe in the initialization function of the library.
- Constants classes shouldn't be inheriting from other static classes of constants.

File: LongoMatch.Services/TemplatesService.cs [hunk 4]
```diff
@@ -236,7 +238,8 @@ public TeamTemplatesProvider (IStorage storage) : base (storage)
 		}
 	}
 
-	public class CategoriesTemplatesProvider : TemplatesProvider<Dashboard>, ICategoriesTemplatesProvider
+	//public class CategoriesTemplatesProvider : TemplatesProvider<Dashboard>, ICategoriesTemplatesProvider
```
Comment (vguzmanp) [inline LongoMatch.Services/TemplatesService.cs]: Commented code

File: LongoMatch.Plugins.Stats/gtk-gui/LongoMatch.Plugins.Stats.SubCategoryViewer.cs [hunk 5]
```diff
@@ -43,9 +49,7 @@ protected virtual void Build ()
 			global::Gtk.Box.BoxChild w2 = ((global::Gtk.Box.BoxChild)(this.hbox1 [this.GtkScrolledWindow]));
 			w2.Position = 0;
 			// Container child hbox1.Gtk.Box+BoxChild
-			this.plotter = new global::LongoMatch.Plugins.Stats.Plotter ();
-			this.plotter.Events = ((global::Gdk.EventMask)(256));
-			this.plotter.Name = "plotter";
+			this.plotter = null;
```
Comment (vguzmanp) [inline LongoMatch.Plugins.Stats/gtk-gui/LongoMatch.Plugins.Stats.SubCategoryViewer.cs]: we need to regenerate this

File: LongoMatch.Drawing.Cairo/CairoBackend.cs [hunk 6]
```diff
@@ -18,16 +18,16 @@
...
-using LineStyle = LongoMatch.Core.Common.LineStyle;
-using Point = LongoMatch.Core.Common.Point;
+using VAS.Core.Common;
+using Color = VAS.Core.Common.Color;
+using FontAlignment = VAS.Core.Common.FontAlignment;
+using FontSlant = VAS.Core.Common.FontSlant;
+using FontWeight = VAS.Core.Common.FontWeight;
+using Image = VAS.Core.Common.Image;
+using LineStyle = VAS.Core.Common.LineStyle;
+using Point = VAS.Core.Common.Point;
```
Comment (vguzmanp) [inline LongoMatch.Drawing.Cairo/CairoBackend.cs]: Redundant aliases

File: LongoMatch.Drawing.Cairo/WidgetWrapper.cs [hunk 7]
```diff
@@ -17,17 +17,18 @@
...
-using CursorType = LongoMatch.Core.Common.CursorType;
+using LongoMatch.Core.Interfaces.Drawing;
+using VAS.Core;
+using VAS.Core.Common;
+using VAS.Core.Interfaces.Drawing;
+using CursorType = VAS.Core.Common.CursorType;
 using GCursorType = Gdk.CursorType;
-using Image = LongoMatch.Core.Common.Image;
+using Image = VAS.Core.Common.Image;
+using Point = VAS.Core.Common.Point;
```
Comment (vguzmanp) [inline LongoMatch.Drawing.Cairo/WidgetWrapper.cs]: Redundant aliases

File: LongoMatch.Drawing/Widgets/PlaysTimeline.cs [hunk 8]
```diff
@@ -208,7 +209,10 @@ void FillCanvas ()
 			}
 			                        
 			foreach (EventType type in project.EventTypes) {
-				tl = new CategoryTimeline (project, project.EventsByType (type), duration,
+				List<TimelineEvent> timelineEventList = project.EventsByType (type);
+				var timelineEventLongoMatchList = new List<TimelineEventLongoMatch> ();
+				timelineEventList.ForEach (x => timelineEventLongoMatchList.Add (x as TimelineEventLongoMatch));
+				tl = new CategoryTimeline (project, timelineEventLongoMatchList, duration,
```
Comment (vguzmanp) [inline LongoMatch.Drawing/Widgets/PlaysTimeline.cs]: This would be cleaner overriding `EventsByType` in ProjectLongoMatch

File: LongoMatch.GUI/Gui/Component/EventsListWidget.cs [hunk 9]
```diff
@@ -82,8 +83,14 @@ void UpdateTeamsModels ()
 		{
 			if (project == null)
 				return;
-			localPlayersList.SetTeam (project.LocalTeamTemplate, project.Timeline);
-			visitorPlayersList.SetTeam (project.VisitorTeamTemplate, project.Timeline);
+
+			var timeline = new ObservableCollection<TimelineEventLongoMatch> ();
+			foreach (var timelineEvent in project.Timeline) {
+				timeline.Add (timelineEvent as TimelineEventLongoMatch);
+			}
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Component/EventsListWidget.cs]: Why is this here? 😮

Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Component/EventsListWidget.cs]: We should override `project.Timeline`

File: LongoMatch.GUI/Gui/Component/PlaysPositionViewer.cs [hunk 10]
```diff
@@ -44,42 +43,47 @@ public PlaysPositionViewer ()
...
 
-		public void LoadProject (Project project, EventsFilter filter)
+		public void LoadProject (ProjectLongoMatch project, EventsFilter filter)
 		{
 			this.project = project;
 			if (project != null) {
+				var timeLine = new ObservableCollection<TimelineEventLongoMatch> ();
+				foreach (var timeLineEvent in project.Timeline) {
+					timeLine.Add (timeLineEvent as TimelineEventLongoMatch);
+				}
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Component/PlaysPositionViewer.cs]: we should override `project.Timeline`

File: LongoMatch.GUI/Gui/GUIToolkit.cs [hunk 11]
```diff
@@ -296,20 +299,20 @@ public Task EditPlay (TimelineEvent play, Project project, bool editTags, bool e
...
 			dialog.Destroy ();
 			return project;
 		}
 
-		public void SelectProject (List<Project> projects)
+		public void SelectProject (List<ProjectLongoMatch> projects)
 		{
 			Log.Information ("Select project");
-			mainWindow.SelectProject (projects);
+			mainWindow.SelectProject (projects.ToList ());
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/GUIToolkit.cs]: Redundant `ToList`

File: LongoMatch.DB/CouchbaseManagerLongoMatch.cs [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+		}
+
+		public override IStorage Add (string name)
+		{
+			// Couchbase doesn't accept uppercase databases.
+			name = SanitizeDBName (name);
+			var storage = Add (name, false);
+			if (storage != null) {
+				VAS.Config.EventsBrokerBase?.EmitDatabaseCreated (name);
+				Config.EventsBroker?.EmitDatabaseCreated (name);
```
Comment (vguzmanp) [inline LongoMatch.DB/CouchbaseManagerLongoMatch.cs]: When we improve the EventsBroker this whole method won't be necessary

File: LongoMatch.GUI/Gui/Component/EventsListWidget.cs (hunk 9 above)
Comment (aameijide) [inline LongoMatch.GUI/Gui/Component/EventsListWidget.cs]: A cleaner way to do this cast is: project.Timeline.Cast<TimelineEventLongoMatch>.ToList()

Maybe is possible to substitute Cast<> with OfType<>

File: LongoMatch.DB/CouchbaseStorageLongoMatch.cs [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+	{
+		Database db;
+		Dictionary<Type, object> views;
+		string storageName;
+		object mutex;
+
+		static CouchbaseStorageLongoMatch ()
+		{
+			DocumentsSerializerHelper.AddTypeTranslation(typeof(TeamTemplate), typeof(Team));
+			DocumentsSerializerHelper.AddTypeTranslation(typeof(TimelineEventLongoMatch), typeof(TimelineEventLongoMatch));
```
Comment (vguzmanp) [inline LongoMatch.DB/CouchbaseStorageLongoMatch.cs]: Why is this type translation needed?
